File: Algorithms/2161. Partition Array According to Given Pivot.ts
function pivotArray(nums: number[], pivot: number): number[] {
    let left: number[] = []
    let right: number[] = []
    let count = 0
    for (let i=0; i<nums.length; i++) {
        if (pivot === nums[i]) count++
        else if (nums[i] < pivot) left.push(nums[i])
        else right.push(nums[i])
    }
    // output 
    let output: number[] = []
    for (var num of left) output.push(num)
    for (let i=0; i<count; i++) output.push(pivot)
    for (var num of right) output.push(num)
    return output
};

File: Algorithms/2161. Partition Array According to Given Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> left, right, output; 
        int count = 0;
        for (int& num : nums) {
            if (num == pivot) count++;
            else if (num < pivot) left.push_back(num);
            else right.push_back(num);
        }
        // output 
        for (int& num : left) output.push_back(num);
        for (int i=0; i<count; i++) output.push_back(pivot);
        for (int& num : right) output.push_back(num);
        return output;
    }
};